src/rules/string-constant-types.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+module.exports = {
+    create(context) {
+        let isFlowAnnotated = false;
+
+        return {
+            Program(node) {
+                isFlowAnnotated = node.comments.some(
+                    (comment) => comment.value.trim() === '@flow'
+                );
+            },
+            ExportNamedDeclaration(node) {
+                if (!isFlowAnnotated) {
+                    return;
+                }
+
+                // E.g. export { default } from './container'
+                if (!node.declaration) {
+                    return;
+                }
+
+                if (node.declaration.kind !== 'const') {
+                    return;
+                }
+
+                const [declaration] = node.declaration.declarations;
+
+                if (!declaration.init || declaration.init.type !== 'Literal') {
+                    return;
+                }
+
+                if (typeof declaration.init.value !== 'string') {
+                    return;
+                }
+
+                if (declaration.id.typeAnnotation) {
+                    return;
+                }
+
+                const maxLiteralLength = 100;
```

Comment:
why is this limitation needed?

Reply:
Flow has a 100 character limit for literals, something I didn't know!

https://flow.org/try/#0MYewdgzgLgBAhhCBTKAVAngByRAXDAcgDEAPAZUxCgBpSiQAnAdzgYBMJaSA1OMASwA2guAHlMUfuGoBhcFAZxgUIgFcoqhjlrrNOcZOlkoIYAGtqx02YCSYNkhKWT5g1LCyAZm1Fgr5rx8wOwcnACFwNkDfNQ0tTlIAaTBrOzdpJJTXdXSPUjD+MFZ0XK5fJFQQVWAAC1LSADkQSuq6iXdnVPtHet1440UoJABzdAIYAF4AKBhZwlIKKi56ZlYOLl4BYTF26TkwBSUVPu1YvQhS-wsrkMdO112PGW9fK+jg7vDI97P4rmSuvUSADslAgQUigwSo8ymAKlVakCmi1ETCbp9enEcAM4ENRgQANxTIA